packages/flame/lib/src/components/text_box_component.dart

```diff
@@ -165,19 +185,14 @@ class TextBoxComponent<T extends TextRenderer> extends PositionComponent {
     c.drawImage(_cache!, Offset.zero, _imagePaint);
   }
 
-  @override
-  void onGameResize(Vector2 gameSize) {
-    super.onGameResize(gameSize);
-    _gameSize = gameSize;
-  }
-
   Future<Image> _redrawCache() {
     final recorder = PictureRecorder();
-    final c = Canvas(recorder, _gameSize.toRect());
+    final c = Canvas(recorder, size.toRect());
```

Comment:
on most cases, gameSize will just be bigger than size and this would just work. however:

1. due to my other mistake, it was zero (should be late and NEVER initialized with zero anyway)
2. this way it doesn't even use gameSize so simpler code
3. it's just more correct as technically this could be bigger than the screen size
4. doesn't allocate bigger image unecessarily

Honestly I am not sure why I ever wrote it using game size, since I clearly understood I had the actual size handy two lines below. go figure!